geode-core/src/main/java/org/apache/geode/management/internal/configuration/mutators/ConfigurationManager.java

```diff
@@ -31,12 +31,12 @@
  * type {@link CacheElement}, which represents the configuration change.
  */
 @Experimental
-public interface ConfigurationManager<T extends CacheElement, R extends RuntimeCacheElement> {
+public interface ConfigurationManager<T extends CacheElement> {
   void add(T config, CacheConfig existing);
 
   void update(T config, CacheConfig existing);
 
   void delete(T config, CacheConfig existing);
 
-  List<R> list(T filterConfig, CacheConfig existing);
+  List<? extends RuntimeCacheElement> list(T filterConfig, CacheConfig existing);
```

Comment:
I feel that this may be a step in the wrong direction.  Returning a list with a wildcard type is frowned upon (Effective Java 3rd edition, Item 31 says in bold: **Do not use bounded wildcard types as return types**).  We can get away with it in a lot of tests because they only validate the response as a RuntimeCacheElement, but this change will force any real code that wants to interact with the response as a RuntimeRegionConfig to cast it or do something giving a raw type warning.  Already this change has forced some new raw type warnings in LocatorClusterManagementServiceTest.java...

Reply:
yeah, but that's just some tests. I would rather do this than having to declare multiple T,R. It's confusing to the users. At least for now, I don't see the need to complicate this interface this much.